components/templates/McGonagall/McGonagall.tsx

```diff
@@ -241,8 +408,11 @@ class McGonagall extends React.Component {
    * @param {object} state state machine
    * @returns {object} step
    */
-  getMatchingStep(allSteps, state) {
-    return allSteps.find((step) => state.matches(step.name));
+  getMatchingStep(
+    allSteps: Step[],
+    state: State<TContext, McGonagallEvent, StateSchema, any>,
+  ): Step {
+    return allSteps.find((step) => state.matches(step.name))!;
```

Comment:
Similar to my earlier comment about throwing an explicit error, based on the usage of the results from `getMatchingStep` I think something like the below will help developers catch bugs that should only happen in development and will give explicit errors for a thing that might otherwise be difficult to track down if it happens when deployed. `state.value` may not be the best piece of information to use in the error message but I think this suggestion conveys intent.

```suggestion
    const result = allSteps.find((step) => state.matches(step.name));
    if (result === null || typeof result === 'undefined') {
      throw new Error(`Unable to find step matching ${state.value}`);
    }
    return result;
```